internal/store/store.go

```diff
@@ -494,12 +518,18 @@ func (bs *BlockStore) SaveBlock(block *types.Block, blockParts *types.PartSet, s
 	}
 
 	// Save seen commit (seen +2/3 precommits for block)
-	pbsc := seenCommit.ToProto()
+	pbsc := seenCommit.StripExtensions().ToProto()
 	seenCommitBytes := mustEncode(pbsc)
 	if err := batch.Set(seenCommitKey(), seenCommitBytes); err != nil {
 		panic(err)
 	}
 
+	pbec := seenCommit.ToProto()
+	extCommitBytes := mustEncode(pbec)
+	if err := batch.Set(extCommitKey(height), extCommitBytes); err != nil {
```

Comment:
Well that does depend on the pruning strategy - what is the pruning strategy that we're going with? :slightly_smiling_face:

Reply:
@williambanfield 's consideration makes sense to me, although I would propose to defer the discussion/design/implementation of this: currently there is no pruning strategy. So IMO the best moment to reflect on this would be when we set out to implementing storage optimizations (whether simply pruning, or eventually going for solution 3 in RFC 017).

Reply:
We probably do need to make a decision on this before we cut a release though - at least where to store the extensions, not the pruning strategy. Otherwise we may end up needing tooling to migrate from one storage approach to another.

I'll investigate what it would take to store vote extensions separately to blocks.

Reply:
yup!

Reply:
yes